src/clj/puppetlabs/services/ca/certificate_authority_core.clj

```diff
@@ -236,29 +235,27 @@
       (ca/get-certificate-statuses settings)
       (as-json-or-pson context))))
 
-(schema/defn routes
+(schema/defn ^:always-validate web-routes :- comidi/BidiRoute
   [ca-settings :- ca/CaSettings]
-  (compojure/context "/:environment" [environment]
-    (compojure/routes
-      (ANY "/certificate_status/:subject" [subject]
-        (certificate-status subject ca-settings))
-      (ANY "/certificate_statuses/:ignored-but-required" [do-not-use]
-        (certificate-statuses ca-settings)))
-      (GET "/certificate/:subject" [subject]
-        (handle-get-certificate subject ca-settings))
-      (compojure/context "/certificate_request/:subject" [subject]
-        (GET "/" []
-          (handle-get-certificate-request subject ca-settings))
-        (PUT "/" {body :body}
-          (handle-put-certificate-request! subject body ca-settings)))
-      (GET "/certificate_revocation_list/:ignored-node-name" []
-        (handle-get-certificate-revocation-list ca-settings))))
+  (comidi/context ["/" :environment]
+    (ANY ["/certificate_status/" :subject] [subject]
+         (certificate-status subject ca-settings))
+    (ANY ["/certificate_statuses/" :ignored-but-required] []
```

Comment:
No, we definitely don't need it.  I am not sure if there if Compojure requires it to be there - if not, it's only there because I didn't know it could be omitted.